github/search.go

```diff
@@ -17,6 +20,42 @@ import (
 // GitHub API docs: http://developer.github.com/v3/search/
 type SearchService service
 
+type SearchQualifiers struct {
+	In        string     `qual:"in"`
+	Size      uint64     `qual:"size"`
+	Forks     uint64     `qual:"forks"`
+	Fork      string     `qual:"fork"`
+	Created   *time.Time `qual:"created"`
+	Pushed    *time.Time `qual:"pushed"`
+	Updated   *time.Time `qual:"pushed"`
+	Merged    *time.Time `qual:"merged"`
+	Closed    *time.Time `qual:"closed"`
+	User      string     `qual:"user"`
+	Repo      string     `qual:"repo"`
+	Repos     uint64     `qual:"repos"`
+	Language  string     `qual:"language"`
+	Stars     uint64     `qual:"stars"`
+	Path      string     `qual:"path"`
+	FileName  string     `qual:"filename"`
```

Comment:
I'm not yet sure what to say to this PR at a high level (still thinking and waiting for others' comments), but at a low level, this should be `Filename`, not `FileName`. See similar fields:

- https://godoc.org/github.com/google/go-github/github#CommitFile.Filename
- https://godoc.org/github.com/google/go-github/github#GistFile.Filename
- https://godoc.org/github.com/google/go-github/github#GistFilename